pkg/kubectl/cmd/get.go

```diff
@@ -476,5 +481,10 @@ func RunGet(f *cmdutil.Factory, out io.Writer, errOut io.Writer, cmd *cobra.Comm
 	if printer != nil && lastMapping != nil {
 		cmdutil.PrintFilterCount(filteredResourceCount, lastMapping.Resource, errOut, filterOpts)
 	}
+
+	// output empty list warning when no resources to display
+	if len(objs) == 0 {
+		fmt.Fprintf(errOut, "%s\n", "No resources found.")
```

Comment:
Check for `len(infos)` here, and move both checks right after `infos` is created. 
We can also return right away after no resource found (optional).